src/lib/comms/include/sol-http.h

```diff
@@ -236,9 +250,26 @@ sol_http_param_init(struct sol_http_param *params)
 
 bool sol_http_param_add(struct sol_http_param *params,
     struct sol_http_param_value value) SOL_ATTR_WARN_UNUSED_RESULT;
+bool sol_http_param_add_copy(struct sol_http_param *params, struct sol_http_param_value value);
 void sol_http_param_free(struct sol_http_param *params);
 
-int sol_http_escape_string(char **escaped, const char *value);
+int sol_http_encode_slice(struct sol_buffer *buf, const struct sol_str_slice value);
```

Comment:
document all these functions... in the header! (also valid for those you didn't do, but do those in a different commit later)